### PR TITLE
Zoom and enhance

### LINKED ISSUE #1217: Allow users to zoom
via @cheshire137 

> wondering if font size adjustment is something that's been considered. I want to increase the font size in the app. Slack allows this via Cmd-+, like in a browser.

I agree that this is good to have, especially from an accessibility stance.

Given that there are instances where we hardcode row heights and font sizes, we might want to discuss what are the limitations that we would run into from both an experience and a technical perspective. For example, if you push Slack's zoom to 150%, the text gets cut off because it doesn't wrap and you can't scroll horizontally.

### PR DESCRIPTION
**Pull request by niik**
wt0fxn

This enables zooming of the app along the same zoom scale that Chrome supports (110%, 125%, 150%, 175% and 200%)

cc #1646, #1547 

Fixes #1217

### CODE REVIEW

**Comment by niik** [comment]
This is ready for review now. The only remaining issue we have is that certain zoom levels on Windows (or rather on 100% DPI monitors) will cause a rounding issue with our file list items that cause the list item border to disappear. This is temporary and goes away as soon as you resize the sidebar or switch between history/changes so I don't think it's worth holding up this PR for. It might fix itself with newer version of Chromium. A workaround would be for us to bump the list item height from 29px to 30px such that the rounding error doesn't occur but I didn't want to do that without talking to @donokuda first as he put in a lot of time to arrive at that 29px height.

File: app/src/lib/app-state.ts [hunk 1]
```diff
@@ -48,6 +48,12 @@ export interface IAppState {
   readonly windowState: WindowState
 
   /**
+   * The current zoom factor of the Window represented as a fractional number
```
Comment (joshaber) [inline app/src/lib/app-state.ts]: Lol can Window be lowercased?

File: app/styles/ui/window/_zoom-info.scss [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    min-width: 100px;
+    background: rgba($gray-900, 0.6);
+    color: $gray-000;
+    text-align: center;
+    border-radius: 100px;
+    font-size: var(--font-size-md);
+    font-weight: var(--font-weight-semibold);
+
+    // This is so sad but CodeMirror uses z-indexes like crazy
+    // and we need to make sure we stay above all of that.
```
Comment (joshaber) [inline app/styles/ui/window/_zoom-info.scss]: :hurtrealbad:

**Comment by niik** [comment]
🐨

**Pull request merged**